### DISCUSSION TITLE
v2.4.0

### DISCUSSION THREAD

**Discussion by STRRL** [chaos-mesh/chaos-mesh#3671] [General]
# v2.4.0 Release Notes

This release introduces several new features, enhancements, bug fixes for Chaos Mesh.

## New Features

- Add support for PhysicalMachine in UI #3624

## Enhancements
- Change e2e config settings to append "pause image" args. #3567
- Update display of disabled scope in UI #3621
- Make the Scope render conditionally #3622
- Use int64 to restore latency for BlockChaos #3638
- Remove CRD v1beta1 support #3630

## Fixes

- Protect chaos available namespace and filter namespaces if needed #3473
- Respect flag enableProfiling and do not register profiler endpoints when it's false #3474
- Fix the blank screen after creating chaos experiment with "By YAML" #3489
- Update hint text about the manual token generating process for Kubernetes 1.24+ #3505
- Fix NetworkChaos fail with @ifxx in the device #3605
- Fix IOChaos containerNames field in UI #3533
- Fix BlockChaos can't show Chinese name. #3536
- Add omitempty JSON tag to optional fields of the CRD objects. #3531
- Fix "sidecar config" e2e test cases run failed in some scenario.#3564
- Fix recover bug when setting force recover to true #3578
- Fix Integration test with bumping kubectl version. #3589
- Fix generate token failed on chaos dashboard #3595
- Remove limit action of BlockChaos in the dashboard #3655
- Sync PhysicalMachineChaos to API client and forms #3660
- Dashboard: Fix rbac.yaml for token generation verbs/resources #3370
- Refine logging, remove the usage of klog in event recorder #3629

## Others

- Replace io/ioutil package with os package. #3539
- Refine logging in pkg/dashboard/apiserver/event, moved package level variable log into struct Service as a field. #3528
- Refine logging in pkg/dashboard/apiserver/auth/gcp, moved package level variable log into struct Service as a field #3527

<hr /><em>This discussion was created from the release <a href='https://github.com/chaos-mesh/chaos-mesh/releases/tag/v2.4.0'>v2.4.0</a>.</em>

**Comment by fbi007130** [comment]
Chaosd has been deployment in all workernode ,but no physical machines was found in page of experiments creating.
The option Use new PhysicalMachine CRD is enable and selector has been set in valid value.
Is any other configuration should be set?

**Reply by STRRL** [reply]
Chaos Mesh would NOT create PhysicalMachine for each Kubernetes Node automatically, you should manually configure nodes with `chaosd`: https://chaos-mesh.org/docs/simulate-physical-machine-chaos/

**Reply by fbi007130** [reply]
I see and the chaosd has already  been installed in all kubernetes node ,'No physical machines found.' is display after scope was set.
![image](https://user-images.githubusercontent.com/3862344/192413933-8cc29d95-6147-4d50-9cb3-38b965c50bc6.png)
:~> /usr/local/chaosd-latest-linux-amd64/chaosd version
Chaosd Version: version.Info{GitVersion:"main-gf6d3a9fb1e1b0e", GitCommit:"f6d3a9fb1e1b0e90e6667d0fe87bfc572b78f90e", BuildDate:"2022-09-20T10:01:02Z", GoVersion:"go1.18.2", Compiler:"gc", Platform:"linux/amd64"}
